dual_net.py

```diff
@@ -134,6 +139,45 @@ def should_log(i):
                     logger.add_summary(weight_summaries, global_step)
             self.save_weights()
 
+    def validate(self, tf_records, batch_size=128, init_from=None, num_steps=1000):
+        """Compute only the cost components for a set of tf_records, ideally a
+        holdout set, and report them to an 'eval' subdirectory of the logs.
+        """
+        cost_tensor_names = ['policy_cost', 'value_cost', 'l2_cost',
+                             'combined_cost']
+        if self.logdir is None:
+            print("Error, trainer not initialized with a logdir.")
+            return
+
+        logdir = self.logdir + '-eval'
+
+        with self.sess.graph.as_default():
+            input_tensors = preprocessing.get_input_tensors(
+                batch_size, tf_records, shuffle_buffer_size=10000, filter_amount=0.5)
```

Comment:
should this be using SHUFFLE_BUFFER_SIZE?

Reply:
no; i can make the constant TRAIN_SHUFFLE_BUFFER_SIZE but that seems a little verbose.

Reply:
There's no consequences to not shuffling the validation data at all, because we're only computing the overall average error.